### ISSUE TITLE
[Feature Request] Add support for Private Tokens

### ISSUE THREAD

**Issue by colinsidi** [dynatrace-oss/terraform-provider-dynatrace#44]
Currently only admin tokens are allowed to be used, which limits usability of this tool to only admins
Adding ability to use private tokens created by users would extend its functionality

Its unlikely that we would give end-users admin tokens since that defeats any security controls we are using

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hi @colinsidi 

We've been discussing this topic already internally. A solution for your feature request is planned to come with one of the upcomming releases.

**Comment by bala002211** [comment]
@Dynatrace-Reinhard-Pilz - Any update on this?
We are actively looking forward to this feature.
We are on hold until we get this feature to take the "terraform module" to a wider audience

**Comment by Dynatrace-Reinhard-Pilz** [comment]
Hello @bala002211 

The challenge behind this feature is, that Personal Access Tokens focus primarily on the (Settings API)[https://www.dynatrace.com/support/help/dynatrace-api/environment-api/settings/] of Dynatrace.
Some of the currently supported settings are already available via Settings API (SLOs, Span Attributes, ...), some are still available exclusively via (Configuration API)[https://www.dynatrace.com/support/help/dynatrace-api/configuration-api/]. Access to the Configuration API is unfortunately not possible via Pesonal Access Tokens (you cannot assign permissions like ReadConfiguration, WriteConfiguration to a PAT).
So, in short, I'm bound to keep this feature on hold until the new Settings API of Dynatrace offers the remaining settings. I'll make a few inquiries for an ETA here and keep you updated.

best regards,
Reinhard

**Comment by kishikawa12** [comment]
Hi @kannanbalaji,

A significant portion of the Dynatrace configuration is now available via Settings 2.0 and the provider is up-to-date on available resources for those schemas. 

A personal token can be used with the provider but note that it is bound to the users permissions and not all scopes are available ([doc link](https://www.dynatrace.com/support/help/manage/access-control/access-tokens/personal-access-token)). The [Migration Helper](https://github.com/dynatrace-oss/terraform-provider-dynatrace/blob/main/documentation/Terraform%20-%20Migration%20Helper.xlsm) spreadsheet can be used as reference for what resources are available with a personal token. Under the API column, anything under `Settings 2.0` or `Environment v2` (except `dynatrace_credentials`), a total of 186 resources (current) should work with a personal token. 

I hope that helps. I'll close this ticket for the time being, please feel free to reach out if you have any questions.

Thanks!
